### PR TITLE
Adding translations feature to AMO

### PR DESCRIPTION
**Pull request by spoji**
This change-set add back the feature we lost with https://github.com/mozilla/addons-server/issues/4990

It also add translation features directly inside the listing page for summaries and descriptions.

To be able to test this out, I can provide you a key. Just ask me privately by email or IRC.

### CODE REVIEW

**Comment by spoji** [comment]
To be able to test this out, I can provide you a key. Just ask me privately by email or IRC.

File: webextension/background/translation.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+var tokenttl = 1000*60*10;
+var tokenTimeout = 0;
+
+sdk.runtime.onMessage.addListener((data, sender, sendReply) => {
+    if (data.action !== "translate") {
+        return;
+    }
+
+    chrome.storage.local.get({ "translation-secret-key": "" }, (prefs) => {
+        if (!prefs["translation-secret-key"] || prefs["translation-secret-key"] == "") {
```
Comment (kewisch) [inline webextension/background/translation.js]: `!prefs["translation-secret-key"]` should cover both null and empty string.

File: webextension/background/translation.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            sendReply({ error: 'Secret key not set' });
+            return;
+        }
+
+        if(tokenTimeout < Date.now()) {
+            var xhr = new XMLHttpRequest();
```
Comment (kewisch) [inline webextension/background/translation.js]: Can you use the `fetch()` API instead? Makes things much nicer to read.

File: webextension/background/translation.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            xhr.onreadystatechange = function() {
+                if (xhr.readyState == 4) {
+                    authToken = xhr.responseText;
+                    tokenTimeout = Date.now() + tokenttl;
+                    translate(data.text, sendReply);
+                }
+            };
+            xhr.send();
+        }
+        else {
```
Comment (kewisch) [inline webextension/background/translation.js]: There should be an .eslintrc in the base. Can you run eslint to make sure style is ok?

File: webextension/content/translate/review.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            if(data.error) {
+                clickedNode.text("Error : Key not set. See addon's options page.");
+            }
+            else {
+                clickedNode.remove();
+                textNode.text(data.text);
+            }
+        });  
+    }
+});
```
Comment (kewisch) [inline webextension/content/translate/review.js]: Newlines at EOF please :)

File: webextension/manifest.json [hunk 5]
```diff
@@ -82,6 +85,13 @@
       "js": ["content/tabclose/find_addonid.js"],
       "run_at": "document_end"
     },{
+      "matches": [ "https://addons.mozilla.org/en-US/editors/queue/reviews*" ],
+      "js": [
+        "libs/jquery-3.1.1.min.js",
+        "content/translate/review.js"
```
Comment (kewisch) [inline webextension/manifest.json]: I've been using `review.js` for the review pages, maybe moderatedreviews.js ?

**Comment by spoji** [comment]
Good catch! I see the eslint issues now that I'm back on my work machine. My linux laptop during the SF AllHands was not properly configured. I'll do the requested changes and create a new pull request.

**Pull request merged**